### ISSUE TITLE
🐛 BUG: Git is not found when using npm init astro on WSL2

### ISSUE THREAD

**Issue by hmvp** [withastro/astro#2144]
### What version of `astro` are you using?

0.21.11

### What package manager are you using?

npm

### What operating system are you using?

Linux on WSL2 (Ubuntu 20.04.3 LTS)

### Describe the Bug

```sh
$ npm init astro
npx: installed 16 in 4.911s

Welcome to Astro! (create-astro v0.6.9)
If you encounter a problem, visit https://github.com/withastro/astro/issues to search or file a new issue.

> Prepare for liftoff.
> Gathering mission details...
✔ Directory not empty. Continue [force overwrite]? … yes
✔ Which app template would you like to use? › Starter Kit (Generic)
✔ Which frameworks would you like to use? › 
> Copying project files...
could not find commit hash for latest
This seems to be an issue with degit. Please check if you have 'git' installed on your system, and install it if you don't have (https://git-scm.com).
If you do have 'git' installed, please file a new issue here: https://github.com/withastro/astro/issues
$ which git
/usr/bin/git
```

Might be related: I am on a machine without direct internet (npm packages come from a npmjs proxy)

### Link to Minimal Reproducible Example

n/a

**Comment by Raia17** [comment]
I have the same problem, but I'm using Windows instead

### **Terminal message**
---
> could not find commit hash for latest
> This seems to be an issue with degit. Please check if you have 'git' installed on your system, and install it if you don't have (https://git-scm.com).
> If you do have 'git' installed, please file a new issue here: https://github.com/withastro/astro/issues
> npm ERR! code 1
> npm ERR! path C:\Users\ruben\Desktop\my-astro-site
> npm ERR! command failed
> npm ERR! command C:\WINDOWS\system32\cmd.exe /d /s /c create-astro
> npm ERR! A complete log of this run can be found in:
> npm ERR!     C:\Users\ruben\AppData\Local\npm-cache\_logs\2021-12-18T19_34_50_101Z-debug.log
---

Does anybody know how to fix it?

**Comment by warrengroom** [comment]
I'm experiencing the exact same issue on Windows. I also have git installed.

**Comment by jonathantneal** [comment]
Thanks for reporting, everyone. I’m unsure if this has been fixed since it was last reported, so I’m assigning this out to get tested and verified.

I know Windows Subsystem and Windows itself are different creatures, but from these reports 'Windows' does seem to be the common denominator.

**Comment by Mikkel-T** [comment]
Hello @hmvp @Raia17 and @warrengroom, do you all still have this issue? And if yes, could you please try to run the command with the `--verbose` flag (https://github.com/withastro/astro/tree/main/packages/create-astro#debugging)?

**Comment by warrengroom** [comment]
Hey @Mikkel-T, I have just tried again and this is what I'm seeing:
PS C:\ASTRO\AST003> npm init astro my-astro-project -- --verbose
Need to install the following packages:
  create-astro
Ok to proceed? (y) y
node:internal/errors:464        
    ErrorCaptureStackTrace(err);
    ^

Error [ERR_MODULE_NOT_FOUND]: Cannot find package 'C:\Users\warre\AppData\Local\npm-cache\_npx\57e9a3bcc0672894\node_modules\node-fetch\' imported from C:\Users\warre\AppData\Local\npm-cache\_npx\57e9a3bcc0672894\node_modules\create-astro\dist\index.js
    at new NodeError (node:internal/errors:371:5)
    at legacyMainResolve (node:internal/modules/esm/resolve:335:9)
    at packageResolve (node:internal/modules/esm/resolve:877:14)
    at moduleResolve (node:internal/modules/esm/resolve:929:18)
    at defaultResolve (node:internal/modules/esm/resolve:1044:11)
    at ESMLoader.resolve (node:internal/modules/esm/loader:422:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:222:40)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:76:40)
    at link (node:internal/modules/esm/module_job:75:36) {
  code: 'ERR_MODULE_NOT_FOUND'
}
npm ERR! code 1
npm ERR! path C:\ASTRO\AST003
npm ERR! command failed
npm ERR! command C:\WINDOWS\system32\cmd.exe /d /s /c create-astro "my-astro-project" "--verbose"

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\warre\AppData\Local\npm-cache\_logs\2022-01-23T11_44_55_219Z-debug.log

**Comment by Mikkel-T** [comment]
Hey @warrengroom, can you please join the discord and open a support thread so that we can try to debug this? https://astro.build/chat

**Comment by Mikkel-T** [comment]
Hey @warrengroom, can you please join the discord and open a support thread so that we can try to debug this? https://astro.build/chat

**Comment by Purus** [comment]
Trying astro first time today and got this  issue. Any solutions please to overcome? I am using win10 command prompt with git already installed.

**Comment by Mikkel-T** [comment]
@Purus can you please try to run the command with the `--verbose` flag (https://github.com/withastro/astro/tree/main/packages/create-astro#debugging)?

**Comment by matthewp** [comment]
Can anyone provide the Windows version they are running on when they run into this issue?

**Comment by TheMartes** [comment]
Same issue here but on MacOS. Git is installed properly, exported in path, using it without no issues at all.

Running on Node 14.20 I get this:
```
01:44:20 AM [debug] Command failed: git ls-remote https://github.com/withastro/astro
fatal: not a git repository: /Users/martes/../.git/modules/secrets

✖ ■■▶ Copying project files...
01:44:20 AM [debug] DegitError: could not find commit hash for latest
    at Degit._cloneWithTar (/Users/martes/.npm/_npx/29527/lib/node_modules/create-astro/node_modules/degit/dist/index-688c5d50.js:14258:10)
    at async Degit.clone (/Users/martes/.npm/_npx/29527/lib/node_modules/create-astro/node_modules/degit/dist/index-688c5d50.js:14096:4)
    at async main (file:///Users/martes/.npm/_npx/29527/lib/node_modules/create-astro/dist/index.js:115:7) {
  code: 'MISSING_REF',
  ref: 'latest'
}
could not find commit hash for latest
This seems to be an issue with degit. Please check if you have 'git' installed on your system, and install it if you don't have (https://git-scm.com).
If you do have 'git' installed, please run this command with the --verbose flag and file a new issue with the command output here: https://github.com/withastro/astro/issues
```

And Node 16:
```
npm timing command:create Completed in 15684ms
npm verb stack Error: command failed
npm verb stack     at ChildProcess.<anonymous> (/Users/martes/.nvm/versions/node/v16.14.0/lib/node_modules/npm/node_modules/@npmcli/promise-spawn/index.js:64:27)
npm verb stack     at ChildProcess.emit (node:events:520:28)
npm verb stack     at maybeClose (node:internal/child_process:1092:16)
npm verb stack     at Process.ChildProcess._handle.onexit (node:internal/child_process:302:5)
npm verb cwd /Users/martes/Code
npm verb Darwin 21.4.0
npm verb argv "/Users/martes/.nvm/versions/node/v16.14.0/bin/node" "/Users/martes/.nvm/versions/node/v16.14.0/bin/npm" "create" "astro@latest" "--verbose"
npm verb node v16.14.0
npm verb npm  v8.3.1
npm ERR! code 1
npm ERR! path /Users/martes/Code
npm ERR! command failed
npm ERR! command sh -c create-astro
npm verb exit 1
npm timing npm Completed in 15878ms
npm verb code 1

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/martes/.npm/_logs/2022-08-10T23_42_59_025Z-debug-0.log
```

I'm using `create-astro v1.0.0`, and macOS 12.3

**Comment by mattheworiordan** [comment]
I had this same issue and realised it was in fact blocked by `git` needing to authenticate with my SSH key i.e. there was a prompt in the background requesting a password.  It seems the STDOUT was swallowed by the `npm create` command.

It would stall at this point:

![MO screenshot 2022-08-12 at 20 56 06](https://user-images.githubusercontent.com/43789/184443806-38efe6bd-b94c-4be4-87c2-160c2c5bb866.png)

**Comment by lexmin0412** [comment]
Same issue with `--verbose`

<img width="946" alt="image" src="https://user-images.githubusercontent.com/24385370/184693266-72a95ff9-65da-45d6-a24b-60db184a112f.png">

my environment:

- system macOS Monterey 12.5
- node v16.14.0
- create-astro v1.0.0

**Comment by TheMartes** [comment]
@mattheworiordan I can confirm I'm using SSH auth for my git. Have you found a solution?

**Comment by TheMartes** [comment]
As I was researching this seems to be a general issue with NPM for those who uses SSH. There's an open issue here: https://github.com/npm/cli/issues/2610 which is still far from being resolved and it's a huge thread.

Some folks might be interested in these workarounds to get it working. 
```
git config --global url."https://github.com/".insteadOf git@github.com:
git config --global url."https://".insteadOf git://
```

Also I've tried `yarn`, `npm` and `pnpm`, all of them seems to have same problem so I believe it's core problem of node. Also both npm 6 and 7 are not working.

**Comment by matthewp** [comment]
Thanks for the workaround @TheMartes. Going to consider this closed since there is nothing we can do on the Astro side.
